pom.xml

```diff
@@ -69,12 +70,61 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials</artifactId>
-      <version>1.9.4</version>
+      <version>2.1.4</version>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>token-macro</artifactId>
-      <version>1.9</version>
+      <version>2.0</version>
+    </dependency>
+    <dependency>
+        <groupId>org.jenkins-ci.plugins.workflow</groupId>
+        <artifactId>workflow-job</artifactId>
+        <version>2.7</version>
+        <optional>true</optional>
```

Comment:
This smells really wrong. You ought not need to have dependencies on `workflow-job` or `workflow-cps` plugins merely to start builds. If you are just using these for testing, which you do not seem to be yet (no functional tests?), then

```xml
<scope>test</scope>
```

AFAICT you only actually need deps on `workflow-step-api` (for `RemoteBuildPipelineStep`) and `script-security` (for `Handle`).

Reply:
Thank you @jglick I did create the following PR https://github.com/sap-production/parameterized-remote-trigger-plugin/pull/19

Reply:
@jglick Hi Jesse, regarding the (asynchronous) StepExecution, I did prepare the following PR https://github.com/sap-production/parameterized-remote-trigger-plugin/pull/21
The only problem that I have is that I would like to return a Handle, and doing this I can only return a boolean, how could I get a return object?